src/main/java/dev/openfga/sdk/api/client/OpenFgaClient.java

```diff
@@ -1130,12 +1131,13 @@ public CompletableFuture<ClientListRelationsResponse> listRelations(
         var options = listRelationsOptions != null
                 ? listRelationsOptions
                 : new ClientListRelationsOptions().maxParallelRequests(DEFAULT_MAX_METHOD_PARALLEL_REQS);
-        if (options.getAdditionalHeaders() == null) {
-            options.additionalHeaders(new HashMap<>());
-        }
-        options.getAdditionalHeaders().putIfAbsent(CLIENT_METHOD_HEADER, "ListRelations");
-        options.getAdditionalHeaders()
-                .putIfAbsent(CLIENT_BULK_REQUEST_ID_HEADER, randomUUID().toString());
+
+        HashMap<String, String> headers = options.getAdditionalHeaders() != null
+                ? new HashMap<>(options.getAdditionalHeaders())
+                : new HashMap<>();
+        headers.put(CLIENT_METHOD_HEADER, "ListRelations");
+        headers.put(CLIENT_BULK_REQUEST_ID_HEADER, randomUUID().toString());
```

Comment:
Why `put` here but `putIfAbsent` above?

Reply:
As `listRelations` calls `clientBatchCheck` we don't want to stomp over the existing values, agree it's kinda awkward having the two differ so maybe I should just match the two?